ClassTrigger.trigger.txt

```diff
@@ -0,0 +1,23 @@
+trigger ClassTrigger on Class__c (before insert, before delete, before update){
+    
+    // trigger code to restrict class deletion if class has more than one female students
+    if(trigger.isDelete){
+        List<AggregateResult> students = [SELECT Class__r.id, COUNT(Name) FROM Student__c WHERE Sex__c = 'female' AND Class__c IN:  Trigger.old GROUP BY Class__r.id HAVING COUNT(Name) > 1];
+        for (AggregateResult everyStudent : students){
+            Trigger.oldMap.get((id)everyStudent.get('Class__r.id')).addError('This class cannot be deleted as it contains more than one female students');    
+        }
+    }
+    
+    // trigger code to delete all related students record if class custom status is changed to reset
+    if(trigger.isUpdate){

```

Comment:
We should check both isBefore and isUpdate before writing any code block that runs only for before update just to ensure that we cover the future scope of some other dev to add afterUpdate action without your knowledge and your code runs on both before and after update then